### PR TITLE
OMN-9856: Adjacency expansion for shared-module changes

### PR DESCRIPTION
**Pull request by jonahgabriel**
## Summary
- Replaces `_resolve` body with adjacency expansion: changed source modules expand through their `reverse_deps` from `test_selection_adjacency.yaml`
- Adds 2 unit tests with exact-set assertions (models → 8 unit dirs; protocols → 4 unit dirs)

Implements OMN-9856, plan task 5.

**Depends on:** #929 (OMN-9866 mapping, which stacks on #928 → #927 → #925). Merge after #929 lands.

Plan: docs/plans/change-aware-ci-omnibase-core.md

## Test plan
- [x] 6 unit tests pass in test_detect_test_paths.py (4 from OMN-9866 + 2 new)
- [x] models change → exactly {tests/unit/<m>/ for m in [models, nodes, contracts, runtime, validation, services, factories, container]} (8 sorted entries)
- [x] protocols change → exactly {tests/unit/<m>/ for m in [protocols, nodes, services, factories]} (4 sorted entries; cli/ MUST NOT appear)
- [x] pre-commit run --all-files passes

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved error handling in module detection with clearer error messages for missing configurations.

* **Documentation**
  * Added comprehensive docstrings to model classes and utility functions.

* **Tests**
  * Enhanced test organization with unit test markers across CI test modules.
  * Expanded validation tests to cover missing module scenarios and edge cases.

### CODE REVIEW

File: scripts/ci/test_selection_loader.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+                if dep not in self.adjacency:
+                    raise ValueError(
+                        f"adjacency['{module}'].reverse_deps references unknown module '{dep}'"
+                    )
+        return self
+
+
+def load_adjacency_map(path: Path) -> ModelAdjacencyMap:
+    raw = yaml.safe_load(path.read_text(encoding="utf-8"))
+    return ModelAdjacencyMap.model_validate(raw)
```
[the remark this answers is not in the document]
Comment (jonahgabriel) [reply scripts/ci/test_selection_loader.py:55]: Acknowledged — tracking in tech-debt backlog. This is a minor/nitpick finding that does not block merge. Auto-triaged by CodeRabbit auto-triage hook [OMN-6739].

**Comment by jonahgabriel** [comment]
Superseded by #940 (merge commit a80fe01d on 2026-04-27, which bundled Tasks 1-13 into a single squash). Closing as duplicate. See audit: docs/tracking/2026-04-27-change-aware-ci-state-audit.md
